modulemd/include/modulemd-2.0/modulemd-obsoletes.h

```diff
@@ -0,0 +1,254 @@
+
+  MD_OBSOLETES_VERSION_UNSET = 0,
+
+  MD_OBSOLETES_VERSION_ONE = 1,
+
+  MD_OBSOLETES_VERSION_LATEST = MD_OBSOLETES_VERSION_ONE,
+} ModulemdObsoletesVersionEnum;
+
+/**
+ * modulemd_obsoletes_new:
+ * @mdversion: (in): The metadata version of this #ModulemdObsoletes.
+ * @modified: (in): The last modified time represented as a 64-bit integer (such as
+ * 201807011200).
+ * @module_name: (in): The name of the module to which this obsoletes applies.
+ * @module_stream: (in): The name of the module stream to which this obsoletes applies.
+ * @message: (in): A string describing the change, reason, etc.
+ *
+ * Returns: (transfer full): A newly-allocated #ModulemdObsoletes object.
+ * This object must be freed with g_object_unref().
+ *
+ * Since: 2.10
+ */
+ModulemdObsoletes *
+modulemd_obsoletes_new (guint64 mdversion,
+                        guint64 modified,
+                        const gchar *module_name,
+                        const gchar *module_stream,
+                        const gchar *message);
+
+/**
+ * modulemd_obsoletes_copy:
+ * @self: (in): This #ModulemdObsoletes object.
+ *
+ * Create a copy of this #ModulemdObsoletes object.
+ *
+ * Returns: (transfer full): The copied #ModulemdObsoletes object.
+ *
+ * Since: 2.10
+ */
+ModulemdObsoletes *
+modulemd_obsoletes_copy (ModulemdObsoletes *self);
+
+/**
+ * modulemd_obsoletes_validate:
+ * @self: (in): This #ModulemdObsoletes object.
+ * @error: (out): If the object is not valid, it will return the reason.
+ *
+ * This method ensures that the obsoletes is internally consistent for usage
+ * or dumping to YAML. It will be run implicitly prior to emitting YAML. This
+ * is not a complete linter, merely a sanity check that the values are not
+ * impossible.
+ *
+ * Since: 2.10
+ */
+gboolean
+modulemd_obsoletes_validate (ModulemdObsoletes *self, GError **error);
+
+/**
+ * modulemd_obsoletes_get_module_context:
+ * @self: (in): This #ModulemdObsoletes object.
+ *
+ * Returns: (transfer none): The context name to which this #ModulemdObsoletes object applies.
+ *
+ * Since: 2.10
+ */
+const gchar *
+modulemd_obsoletes_get_module_context (ModulemdObsoletes *self);
+
+/**
+ * modulemd_obsoletes_set_module_context:
+ * @self: This #ModulemdObsoletes object.
+ * @module_context: (in) (nullable): The name of the module context to which this obsoletes
+ * applies.
+ *
+ * Since: 2.10
+ */
+void
+modulemd_obsoletes_set_module_context (ModulemdObsoletes *self,
+                                       const gchar *module_context);
+
+/**
+ * modulemd_obsoletes_set_modified:
+ * @self: This #ModulemdObsoletes object.
+ * @modified: (in): The last modified time represented as a 64-bit integer (such as
+ * 201807011200).
+ *
+ * Since: 2.10
+ */
+void
+modulemd_obsoletes_set_modified (ModulemdObsoletes *self, guint64 modified);
+
+/**
+ * modulemd_obsoletes_set_reset:
+ * @self: This #ModulemdObsoletes object.
+ * @reset: (in): Whether to cancel/reset all previously specified obsoletes.
```

Comment:
Please make this documentation more clear. It took me multiple readings to realize that this is setting an attribute of the object rather than clearing out the object.